### ISSUE TITLE
JobCreator breaks on the Harvest algo (1.1.4.patch1)

### ISSUE THREAD

**Issue by amaltaro** [dmwm/WMCore#7938]
Another issue from the run/lumi/event changes. Here is the traceback
```
2017-06-16 10:46:15,888:140481720125184:DEBUG:JobCreatorPoller:Have 0 jobs for workflow amaltaro_DQMHarvesting_RunWhitelist_Agent114_Validation_170615_113529_4323 already in database.
2017-06-16 10:46:15,924:140481720125184:DEBUG:Harvest:Creating EndOfRun harvesting job
2017-06-16 10:46:16,299:140481720125184:ERROR:JobCreatorPoller:Failed to execute JobCreator 
list index out of range

Traceback (most recent call last):
  File "/data/srv/wmagent/v1.1.4.patch1/sw/slc6_amd64_gcc493/cms/wmagent/1.1.4.patch1/lib/python2.7/site-packages/WMComponent/JobCreator/JobCreatorPoller.py", line 427, in algorithm
    self.pollSubscriptions()
  File "/data/srv/wmagent/v1.1.4.patch1/sw/slc6_amd64_gcc493/cms/wmagent/1.1.4.patch1/lib/python2.7/site-packages/WMComponent/JobCreator/JobCreatorPoller.py", line 563, in pollSubscriptions
    wmbsJobGroups = next(jobSplittingFunction)
  File "/data/srv/wmagent/v1.1.4.patch1/sw/slc6_amd64_gcc493/cms/wmagent/1.1.4.patch1/lib/python2.7/site-packages/WMComponent/JobCreator/JobCreatorPoller.py", line 88, in runSplitter
    groups = jobFactory(**splitParams)
  File "/data/srv/wmagent/v1.1.4.patch1/sw/slc6_amd64_gcc493/cms/wmagent/1.1.4.patch1/lib/python2.7/site-packages/WMCore/JobSplitting/JobFactory.py", line 94, in __call__
    self.algorithm(*args, **kwargs)
  File "/data/srv/wmagent/v1.1.4.patch1/sw/slc6_amd64_gcc493/cms/wmagent/1.1.4.patch1/lib/python2.7/site-packages/WMCore/JobSplitting/Harvest.py", line 255, in algorithm
    self.createJobsLocationWise(fileset, True, dqmHarvestUnit, lumiMask, goodRunList)
  File "/data/srv/wmagent/v1.1.4.patch1/sw/slc6_amd64_gcc493/cms/wmagent/1.1.4.patch1/lib/python2.7/site-packages/WMCore/JobSplitting/Harvest.py", line 124, in createJobsLocationWise
    self.createJobByRun(locationDict, location, baseName, harvestType, runDict, endOfRun)
  File "/data/srv/wmagent/v1.1.4.patch1/sw/slc6_amd64_gcc493/cms/wmagent/1.1.4.patch1/lib/python2.7/site-packages/WMCore/JobSplitting/Harvest.py", line 144, in createJobByRun
    self.currentJob['mask'].addRun(fileRun)
  File "/data/srv/wmagent/v1.1.4.patch1/sw/slc6_amd64_gcc493/cms/wmagent/1.1.4.patch1/lib/python2.7/site-packages/WMCore/DataStructs/Mask.py", line 109, in addRun
    firstLumi = run.lumis[0]
IndexError: list index out of range

2017-06-16 10:46:16,344:140481720125184:ERROR:BaseWorkerThread:Error in worker algorithm (1):
```

I'm debugging

**Comment by amaltaro** [comment]
Not sure if this issue started happening after I applied this patch:

but the Harvest algo is basically trying to add a Run object into the Mask object, by passing a lumiless run object:
https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/JobSplitting/Harvest.py#L144

I also saw JobAccountant problems, calling a DAO (AddRunLumi) without lumis... more debugging.

**Comment by amaltaro** [comment]
Just found a problem on this fix: #7937 
Patch is updated and I'll follow up there. Leaving it open for a little while until I make sure that was the problem.

**Comment by amaltaro** [comment]
Flaw on the patch created that cause this issue and one more :) Closing it
